addons/html_editor/static/tests/_helpers/editor.js

```diff
@@ -161,9 +163,17 @@ export async function testEditor(config) {
     // (like an autofocus feature), it would be the role of the autofocus
     // feature to trigger the stageSelection.
     editor.shared.history.stageSelection();
-
     if (config.props?.iframe) {
-        expect("iframe").toHaveCount(1);
+        const selection = editor.document.getSelection();
+        let iframeCount = 1;
+        if (selection.anchorNode) {
+            if (selection.isCollapsed) {
+                iframeCount = isMobileTest ? 2 : 1;
+            } else {
+                iframeCount = 2;
+            }
```

Comment:
Which shall be equivalent to:
```suggestion
            iframeCount = isMobileTest || !selection.isCollapsed ? 2 : 1;
```

And I would also suggest to add a comment on why is iframeCount 2 here.